hedera-node/src/main/java/com/hedera/services/sigs/order/HederaSigningOrder.java

```diff
@@ -541,12 +541,11 @@ private boolean hasNondeprecatedAdminKey(ContractUpdateTransactionBody op) {
 		} else if (targetMustSign) {
 			required = mutable(required);
 			required.add(result.metadata().getKey());
-		}
-
-		if (op.hasKey()) {
-			required = mutable(required);
-			var candidate = asUsableFcKey(op.getKey());
-			candidate.ifPresent(required::add);
+			if (op.hasKey()) {
+				required = mutable(required);
```

Comment:
Since you already guaranteed the `required` list is mutable on L542, this line can be deleted.